card/src/main/java/com/adyen/checkout/card/ui/SocialSecurityNumberInput.kt

```diff
@@ -0,0 +1,77 @@
+package com.adyen.checkout.card.ui
+
+import android.content.Context
+import android.text.Editable
+import android.text.InputType
+import android.text.method.DigitsKeyListener
+import android.util.AttributeSet
+import com.adyen.checkout.components.ui.view.AdyenTextInputEditText
+
+class SocialSecurityNumberInput constructor(
```

Comment:
Can you add some unit tests for this class? There seems to be some complicated logic, especially around `formatString`